test/e2e/fixture_test.go

```diff
@@ -0,0 +1,209 @@
+		Eventually(func(g Gomega) {
+			key := types.NamespacedName{Namespace: bmh.Namespace, Name: bmh.Name}
+			g.Expect(bootstrapClusterProxy.GetClient().Get(ctx, key, &bmh)).To(Succeed())
+			g.Expect(bmh.Status.ErrorType).To(Equal(bmov1.RegistrationError))
+		}, "1m").Should(Succeed())
+	})
+
+	It("should inspect a newly created BMH", func() {
+		By("Creating a secret with BMH credentials")
+		bmcCredentials := corev1.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "bmc-credentials",
+				Namespace: namespace.Name,
+			},
+			StringData: map[string]string{
+				"username": "username",
+				"password": "password",
+			},
+		}
+		err := bootstrapClusterProxy.GetClient().Create(ctx, &bmcCredentials)
+		Expect(err).NotTo(HaveOccurred())
+
+		By("creating a BMH")
+		bmh := bmov1.BareMetalHost{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "fixture-inspect",
+				Namespace: namespace.Name,
+			},
+			Spec: bmov1.BareMetalHostSpec{
+				BMC: bmov1.BMCDetails{
+					Address:         "ipmi://127.0.0.1:5678",
+					CredentialsName: "bmc-credentials",
+				},
+			},
+		}
+		err = bootstrapClusterProxy.GetClient().Create(ctx, &bmh)
+		Expect(err).NotTo(HaveOccurred())
+
+		By("registering the BMH")
+		WaitForBmhInState(ctx, WaitForBmhInStateInput{
+			Client: bootstrapClusterProxy.GetClient(),
+			Bmh:    bmh,
+			State:  bmov1.StateRegistering,
+		}, "1m")
+
+		By("inspecting the BMH")
+		WaitForBmhInState(ctx, WaitForBmhInStateInput{
+			Client: bootstrapClusterProxy.GetClient(),
+			Bmh:    bmh,
+			State:  bmov1.StateInspecting,
+		}, "1m")
+
+		By("finishing the inspection so the BMH becomes available")
+		WaitForBmhInState(ctx, WaitForBmhInStateInput{
+			Client: bootstrapClusterProxy.GetClient(),
+			Bmh:    bmh,
+			State:  bmov1.StateAvailable,
+		}, "1m")
+	})
+
+	It("should provision a BMH when an image is provided", func() {
+		By("Creating a secret with BMH credentials")
+		bmcCredentials := corev1.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "bmc-credentials",
+				Namespace: namespace.Name,
+			},
+			StringData: map[string]string{
+				"username": "username",
+				"password": "password",
+			},
+		}
+		err := bootstrapClusterProxy.GetClient().Create(ctx, &bmcCredentials)
+		Expect(err).NotTo(HaveOccurred())
+
+		By("creating a BMH with image and turning it on")
+		bmh := bmov1.BareMetalHost{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "fixture-provision",
+				Namespace: namespace.Name,
+			},
+			Spec: bmov1.BareMetalHostSpec{
+				Online: true,
+				BMC: bmov1.BMCDetails{
+					Address:         "ipmi://127.0.0.1:5678",
+					CredentialsName: "bmc-credentials",
+				},
+				BootMACAddress: "00:c6:14:04:61:b2",
+				Image: &bmov1.Image{
+					URL:      "http://127.0.0.1/images/demo.qcow2",
+					Checksum: "http://127.0.0.1/images/demo.qcow2.sha256",
+				},
+			},
+		}
+		err = bootstrapClusterProxy.GetClient().Create(ctx, &bmh)
+		Expect(err).NotTo(HaveOccurred())
+
```

Comment:
Let's maybe check that it does not skip inspection? And then we can have a separate test with inspection disabled (if needed)?